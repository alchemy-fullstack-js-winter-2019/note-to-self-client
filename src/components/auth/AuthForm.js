import React from 'react';
import PropTypes from 'prop-types';

function AuthForm({ handleAuth, onChange, email, password, name, phone }) {
  return (
    <form onSubmit={handleAuth.bind(null, email, password, name, phone)} id="signup">
      <fieldset>
        <legend>Sign up</legend>
        <label htmlFor="signup-email"></label>
        <input onChange={onChange} type="email" id="signup-email" placeholder="Email" value={email} required/>
        
        <label htmlFor="signup-password"></label>
        <input onChange={onChange} type="password" id="signup-password" placeholder="Password" value={password}
          required/>
        
        <label htmlFor="name"></label>
        <input onChange={onChange} type="text" id="name" placeholder="Full name" value={name} required/>
        
        <label htmlFor="phone"></label>
        <input onChange={onChange} type="tel" id="phone" value={phone} placeholder="phone"/>
        
        <button type="submit" value="Sign up"></button>
      </fieldset>
    </form>
  );
}

AuthForm.propTypes = {
  handleAuth: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  phone: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired
};

export default AuthForm;
