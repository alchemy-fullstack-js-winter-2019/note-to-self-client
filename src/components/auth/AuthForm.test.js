
import React from 'react';
import { shallow } from 'enzyme';
import AuthForm from './AuthForm';

describe('authform tests', () => {
  let wrapper;
  let createUser;
  beforeEach(() => {
    createUser = jest.fn();
    wrapper = shallow(
      <AuthForm
        email="email"
        password="password"
        name="name"
        phone="phone"
        handleAuth={createUser}
        onChange={jest.fn()}
      />
    );
  });
  it('matches a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls the createUser function on sumbit', () => {
    wrapper.find('form').simulate('submit', {
      preventDefault: jest.fn()
    });

    expect(createUser).toHaveBeenCalled();
  });
});
