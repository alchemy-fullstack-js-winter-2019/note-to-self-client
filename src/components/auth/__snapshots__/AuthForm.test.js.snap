// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authform tests matches a snapshot 1`] = `
<Fragment>
  <form
    id="signup"
    onSubmit={[Function]}
  >
    <fieldset>
      <legend>
        Sign up
      </legend>
      <label
        htmlFor="signup-email"
      />
      <input
        id="email"
        onChange={[MockFunction]}
        placeholder="Email"
        required={true}
        type="email"
        value="email"
      />
      <label
        htmlFor="signup-password"
      />
      <input
        id="password"
        onChange={[MockFunction]}
        placeholder="Password"
        required={true}
        type="password"
        value="password"
      />
      <label
        htmlFor="name"
      />
      <input
        id="name"
        onChange={[MockFunction]}
        placeholder="Full name"
        required={true}
        type="text"
        value="name"
      />
      <label
        htmlFor="phone"
      />
      <input
        id="phone"
        onChange={[MockFunction]}
        placeholder="phone"
        type="tel"
        value="phone"
      />
      <button
        type="submit"
        value="Sign up"
      >
        CREAT USER BUTTOn
      </button>
    </fieldset>
  </form>
</Fragment>
`;
